File: mvp/src/main/java/com/thefuntasty/taste/injection/module/BaseFragmentModule.java
package com.thefuntasty.taste.injection.module;

import android.support.v4.app.Fragment;

import com.thefuntasty.taste.TasteFragment;
import com.thefuntasty.taste.injection.annotation.scope.PerScreen;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class BaseFragmentModule<T extends TasteFragment> {

	private final T fragment;

	public BaseFragmentModule(T fragment) {
		this.fragment = fragment;
	}

	@Provides
	@PerScreen
	public T fragmentT() {
		return fragment;
	}

	@Provides
	@PerScreen
	public Fragment fragment() {
		return fragment;
	}
}

File: mvp/src/main/java/com/thefuntasty/taste/injection/module/BaseFragmentModule.kt
package com.thefuntasty.taste.injection.module

import android.support.v4.app.Fragment
import com.thefuntasty.taste.TasteFragment
import com.thefuntasty.taste.injection.annotation.PerScreen
import dagger.Module
import dagger.Provides

@Module
abstract class BaseFragmentModule<out T : TasteFragment>(private val fragment: T) {

	@Provides
	@PerScreen
	fun fragmentT(): T {
		return fragment
	}

	@Provides
	@PerScreen
	fun fragment(): Fragment {
		return fragment
	}
}
